aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecListItemSpan.kt

```diff
@@ -23,10 +24,52 @@ fun createListItemSpan(nestingLevel: Int,
 open class AztecListItemSpan(
         override var nestingLevel: Int,
         override var attributes: AztecAttributes) : IAztecCompositeBlockSpan {
+    fun toggleCheck() {
+        if (attributes.getValue(CHECKED) == "true") {
+            attributes.setValue(CHECKED, "false")
+        } else {
+            attributes.setValue(CHECKED, "true")
+        }
+    }
+
+    override val startTag: String
+        get() {
+            val shouldReplaceCheckedAttribute = attributes.hasAttribute(CHECKED)
+            return if (shouldReplaceCheckedAttribute) {
+                val sb = StringBuilder()
+                sb.append(TAG)
+                for (i in 0 until attributes.length) {
+                    val localName = attributes.getLocalName(i)
+                    if (localName != CHECKED) {
+                        sb.append(" ")
+                        sb.append(localName)
+                        sb.append("=\"")
+                        sb.append(attributes.getValue(i))
+                        sb.append("\"")
+                    }
+                }
+                sb.append(">")
+                if (shouldReplaceCheckedAttribute) {
+                    sb.append("<input type=\"checkbox\" class=\"task-list-item-checkbox\"")
+                    if (attributes.getValue(CHECKED) == "true") {
+                        sb.append(" checked")
+                    }
+                    sb.append(" /")
+                }
+                return sb.toString()
```

Comment:
This logic prints the missing `<input type="checkbox"/>` element when generating the HTML.

Reply:
good idea, I've added it in aee4f3b3250ed1e27b7870b6ab3fb0e07c0c89f5